cmd/soroban-rpc/internal/methods/get_ledgers.go

```diff
@@ -0,0 +1,251 @@
+	switch {
+	case req.Pagination != nil && req.Pagination.Cursor != "":
+		if req.StartLedger != 0 {
+			return errors.New("startLedger and cursor cannot both be set")
+		}
+	case !isStartLedgerWithinBounds(req.StartLedger, ledgerRange):
+		return fmt.Errorf(
+			"start ledger must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance",
+			ledgerRange.FirstLedger.Sequence,
+			ledgerRange.LastLedger.Sequence,
+		)
+	}
+
+	if req.Pagination != nil && req.Pagination.Limit > maxLimit {
+		return fmt.Errorf("limit must not exceed %d", maxLimit)
+	}
+
+	return IsValidFormat(req.Format)
+}
+
+// LedgerInfo represents a single ledger in the response.
+type LedgerInfo struct {
+	Hash            string `json:"hash"`
+	Sequence        uint32 `json:"sequence"`
+	LedgerCloseTime int64  `json:"ledgerCloseTime"`
+
+	LedgerHeader     string          `json:"headerXdr"`
+	LedgerHeaderJSON json.RawMessage `json:"headerJson,omitempty"`
+
+	LedgerCloseMeta     string          `json:"closeMetaXdr"`
+	LedgerCloseMetaJSON json.RawMessage `json:"closeMetaJson,omitempty"`
+}
+
+// GetLedgersResponse encapsulates the response structure for getLedgers queries.
+type GetLedgersResponse struct {
+	Ledgers               []LedgerInfo `json:"ledgers"`
+	LatestLedger          uint32       `json:"latestLedger"`
+	LatestLedgerCloseTime int64        `json:"latestLedgerCloseTime"`
+	OldestLedger          uint32       `json:"oldestLedger"`
+	OldestLedgerCloseTime int64        `json:"oldestLedgerCloseTime"`
+	Cursor                string       `json:"cursor"`
+}
+
+type ledgersHandler struct {
+	ledgerReader db.LedgerReader
+	maxLimit     uint
+	defaultLimit uint
+}
+
+// NewGetLedgersHandler returns a jrpc2.Handler for the getLedgers method.
+func NewGetLedgersHandler(ledgerReader db.LedgerReader, maxLimit, defaultLimit uint) jrpc2.Handler {
+	return NewHandler((&ledgersHandler{
+		ledgerReader: ledgerReader,
+		maxLimit:     maxLimit,
+		defaultLimit: defaultLimit,
+	}).getLedgers)
+}
+
+// getLedgers fetch ledgers and relevant metadata from DB.
+func (h ledgersHandler) getLedgers(ctx context.Context, request GetLedgersRequest) (GetLedgersResponse, error) {
+	ledgerRange, err := h.ledgerReader.GetLedgerRange(ctx)
+	if err != nil {
+		return GetLedgersResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InternalError,
+			Message: err.Error(),
+		}
+	}
+
+	if err := request.validate(h.maxLimit, ledgerRange); err != nil {
+		return GetLedgersResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidRequest,
+			Message: err.Error(),
+		}
+	}
+
+	start, limit, err := h.initializePagination(request, ledgerRange)
+	if err != nil {
+		return GetLedgersResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	ledgers, err := h.fetchLedgers(ctx, start, limit, request.Format)
+	if err != nil {
+		return GetLedgersResponse{}, err
+	}
+	cursor := strconv.FormatUint(uint64(ledgers[len(ledgers)-1].Sequence), 10)
```

Comment:
keep it simple :laughing: 
```suggestion
	cursor := fmt.Sprintf("%d", ledgers[len(ledgers)-1].Sequence)
```

Reply:
@Shaptic I remembered why I used strconv.FormatUint. Its much faster than fmt.Sprintf - https://medium.com/swlh/bad-go-frivolous-sprintf-2ad28fedf1a0